### PR TITLE
fix bug 1140436: add period url param & select box to revisions dashboard

### PR DESCRIPTION
**Pull request by jamonation**
I think that this needs some i18n polish (because I don't know how that system works in Kuma), but it filters by hour/day/week/month per @a2sheppy's feature request.

### CODE REVIEW

File: kuma/dashboards/forms.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 
 LANG_CHOICES = [('', _('All Locales'))] + settings.LANGUAGES
-
+PERIOD_CHOICES = (('', 'None'), ('hour','Hour'), ('day','Day'), ('week','Week'), ('month','30 days'))
```
Comment (groovecoder) [inline kuma/dashboards/forms.py]: The human-readable option labels need the tower gettext_lazy `_` wrapper. Like the `LANG_CHOICES` tuple above.

**Comment by jamonation** [comment]
Ah that's what I was missing, thanks @groovecoder - I've added the _ wrapper, and switched from a tuple to a list of tuples to be consistent with the LANG_CHOICES above.

File: kuma/dashboards/forms.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 
 LANG_CHOICES = [('', _('All Locales'))] + settings.LANGUAGES
-
+PERIOD_CHOICES = [('', _('None')), ('hour', _('Hour')), ('day', _('Day')), ('week',_('Week')), ('month',_('30 days'))]
```
Comment (jezdez) [inline kuma/dashboards/forms.py]: Mind putting new lines after each element?

File: kuma/dashboards/views.py [hunk 3]
```diff
@@ -48,6 +48,24 @@ def revisions(request):
...
+            if preceding_period == 'week':
+                seconds = 7 * 24 * 60 * 60
+            if preceding_period == 'day':
+                seconds = 24 * 60 * 60
+            if preceding_period == 'hour':
+                seconds = 60 * 60
+            # use the form date if present, otherwise, offset from now
+            end_date = (filter_form.cleaned_data['end_date'] or
+                        datetime.datetime.now())
+            start_date = end_date - datetime.timedelta(seconds=seconds)
```
Comment (jezdez) [inline kuma/dashboards/views.py]: Please use `django.utils.timezone.now()` here instead in preparation for enabling Django's timezone support.

**Comment by jamonation** [comment]
Thanks for the review @jezdez - I've incorporated your suggestions.

A note: this is the wrong branch name - I can open another pull request if that would make things easier than editing merge commit messages.

Last thing - would you like me to make a separate bug & branch for doing timezone migrations from datetime to django.utils version, or is there one already that I could take on as a task?

**Comment by groovecoder** [comment]
Spot-check looks good. Up to you if you want to open another PR or amend the commit message on this one.

A new bug in the "Code Cleanup" component would be great.

**Comment by jamonation** [comment]
Sure, I've amended the message. Just waiting on Travis and then we should be all set?

**Pull request merged**